test/integration/apiserver/oidc/oidc_test.go

```diff
@@ -0,0 +1,512 @@
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"list"},
+				Resources:     []string{"pods"},
+				APIGroups:     []string{""},
+				ResourceNames: []string{},
+			},
+		},
+	}
+	defaultRoleBinding = &rbacv1.RoleBinding{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "RoleBinding"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleBindingName},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup: rbac.GroupName,
+				Kind:     rbacv1.UserKind,
+				Name:     defaultOIDCUsernamePrefix + defaultOIDCClaimedUsername,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "Role",
+			Name:     defaultRBACRoleName,
+		},
+	}
+)
+
+func TestOIDC(t *testing.T) {
+	var tests = []struct {
+		name                    string
+		configureInfrastructure func(t *testing.T) (
+			oidcServer *utilsoidc.TestServer,
+			apiServer *kubeapiserverapptesting.TestServer,
+			signingPrivateKey *rsa.PrivateKey,
+			caCertContent []byte,
+			caFilePath string,
+		)
+		configureOIDCServerBehaviour func(t *testing.T, oidcServer *utilsoidc.TestServer, signingPrivateKey *rsa.PrivateKey)
+		configureClient              func(
+			t *testing.T,
+			restCfg *rest.Config,
+			caCert []byte,
+			certPath,
+			oidcServerURL,
+			oidcServerTokenURL string,
+		) *kubernetes.Clientset
+		asserErrFn func(t *testing.T, errorToCheck error)
+	}{
+		{
+			name:                    "ID token is ok",
+			configureInfrastructure: configureTestInfrastructure,
+			configureOIDCServerBehaviour: func(t *testing.T, oidcServer *utilsoidc.TestServer, signingPrivateKey *rsa.PrivateKey) {
+				idTokenLifetime := time.Second * 1200
+				oidcServer.TokenHandler().EXPECT().Token().Times(1).DoAndReturn(utilsoidc.TokenHandlerBehaviourReturningPredefinedJWT(
+					t,
+					signingPrivateKey,
+					oidcServer.URL(),
+					defaultOIDCClientID,
+					defaultOIDCClaimedUsername,
+					defaultStubAccessToken,
+					defaultStubRefreshToken,
+					time.Now().Add(idTokenLifetime).Unix(),
+				))
+			},
+			configureClient: configureClientFetchingOIDCCredentials,
+			asserErrFn: func(t *testing.T, errorToCheck error) {
+				assert.NoError(t, errorToCheck)
+			},
+		},
+		{
+			name:                    "ID token is expired",
+			configureInfrastructure: configureTestInfrastructure,
+			configureOIDCServerBehaviour: func(t *testing.T, oidcServer *utilsoidc.TestServer, signingPrivateKey *rsa.PrivateKey) {
+				configureOIDCServerToReturnExpiredIDToken(t, 2, oidcServer, signingPrivateKey)
+			},
+			configureClient: configureClientFetchingOIDCCredentials,
+			asserErrFn: func(t *testing.T, errorToCheck error) {
+				expectedErr := &apierrors.StatusError{
+					ErrStatus: metav1.Status{
+						Status:  "Failure",
+						Message: "Unauthorized",
+						Reason:  "Unauthorized",
+						Code:    401,
+					},
+				}
+				assert.True(t, apierrors.IsUnauthorized(expectedErr), expectedErr)
```

Comment:
sorry, my first comment contained a mistake which I later edited. This should be:
`apierrors.IsUnauthorized(errorToCheck), errorToCheck)`.

Remove `expectedErr` completely

Reply:
Thanks! Done